src/utils/flatten-cfn-reference.ts

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+/**
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-join.html
+ */
+interface CfnFnJoin {
+  'Fn::Join': [string, unknown[]];
+}
+
+/**
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html
+ */
+interface CfnFnSub {
+  'Fn::Sub': unknown;
+}
+
+/**
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html
+ */
+interface CfnFnGetAtt {
+  'Fn::GetAtt': [unknown, unknown];
+}
+
+/**
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html
+ */
+interface CfnFnImportValue {
+  'Fn::ImportValue': unknown;
+}
+
+/**
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html
+ */
+interface CfnFnRef {
+  Ref: unknown;
+}
+
+/**
+ * Turns a CloudFormation reference object into a flat string for easy suppression
+ * @param node The reference object
+ * @returns Flattened string
+ */
+export const flattenCfnReference = (reference: unknown): string => {
+  const visit = (node: unknown): string => {
+    if (typeof node === 'undefined') {
+      return '';
+    }
+    if (typeof node === 'string') {
+      // Replace the template syntax that Fn::Sub uses with round brackets for easier suppression
+      return node.replace(/\${/g, '(').replace(/}/g, ')');
```

Comment:
```suggestion
      return node.replace(/\${/g, '<').replace(/}/g, '>');
```
`(` and `)` have special meanings in Regex, might be easier to use `<` and `>` instead

Reply:
Makes sense, will change. 👍